geopandas/geoseries.py

```diff
@@ -208,9 +208,16 @@ def __init__(self, data=None, index=None, crs: Optional[Any] = None, **kwargs):
                         "Non geometry data passed to GeoSeries constructor, "
                         f"received data of dtype '{s.dtype}'"
                     )
-            # try to convert to GeometryArray, if fails return plain Series
+            # extract object-dtype numpy array from pandas Series; with CoW this
+            # gives a read-only array, so we try to set the flag back to writeable
+            data = s.to_numpy()
             try:
-                data = from_shapely(s.values, crs)
```

Comment:
It seems that simply changing `s.values` to `s.array` in the original code also does the trick as well. It also seems to be what is recommended in the docs https://pandas.pydata.org/docs/reference/api/pandas.Series.to_numpy.html

> When you need a no-copy reference to the underlying data, [Series.array](https://pandas.pydata.org/docs/reference/api/pandas.Series.array.html#pandas.Series.array) should be used instead.

Reply:
Okay. Let's go ahead with this.